compiler-core/src/erlang/tests/bit_arrays.rs

```diff
@@ -314,3 +314,19 @@ pub fn main() {
 "
     );
 }
+
+#[test]
+fn unicode_character_encoding_in_bit_array_pattern_segment() {
+    assert_erl!(
+        "
+pub fn main() -> Nil {
+  let foo = <<\"\\u{00A9}foo\":utf8>>
+  let _bits = case foo {
+    <<\"\\u{00A9}\":utf8, rest: bits>> -> rest
+    _ -> foo
+  }
+  Nil
+}
+"
```

Comment:
To avoid escaping the string quotes in the example you can wrap the string in `r##`, makes it easier to read the test:
```rust
r#"
  no need to escape these " \
"#
```